src/viam/examples/modules/simple/main.cpp

```diff
@@ -55,7 +54,7 @@ class Printer : public Generic {
     }
 
     static std::string find_to_print(ResourceConfig cfg) {
-        auto printer_name = cfg.name();
+        auto const& printer_name = cfg.name();
```

Comment:
I think the better change is to say `auto&`. The `const`-ness will be inferred. Plain `auto` doesn't capture references.

https://godbolt.org/z/4h9b5xq9a

Reply:
That's silly of the linter. It already is `const`. But I don't mind if you want to keep the `const` to placate the linter. Still, maybe we should consider turning that off: https://clang.llvm.org/extra/clang-tidy/checks/readability/qualified-auto.html#cmdoption-arg-AddConstToQualified.

Reply:
> maybe we should consider turning that off

I'm not sure CI/our clang-tidy would complain. It might just be my local clang-tidy/IDE. I've left it without `const`, as it does seem redundant.